ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -112,10 +124,61 @@ let networks path (list: string -> string list) =
          | Some pair -> pair :: acc
       ) [] (list path)
   in
-  path
-  |> find_eths
-  |> List.map (fun (path, prefix) -> find_all_ips path prefix)
-  |> List.concat
+  let find_vifs vif_path =
+    let extract_vif acc vif_id = ((extend vif_path vif_id), vif_id) :: acc in  
+    List.fold_left extract_vif [] (list vif_path)
+  in
+  let cmp a b = 
+    let open Pervasives in
```

Comment:
I will check out it tomorrow. I remember I hit a compile error on int_of_string. But I'm not sure 100% now.